### PR TITLE
Fixes for SerenityOS

### PR DESCRIPTION
**Pull request by linusg**
See commit messages and #688 for details.

### CODE REVIEW

File: include/private/gcconfig.h [hunk 1]
```diff
@@ -1083,7 +1083,6 @@ extern int etext[], _end[];
 #  define DATASTART PTR_ALIGN_UP((ptr_t)etext, 0x1000)
 #  define DATAEND ((ptr_t)_end)
 #  define DYNAMIC_LOADING
-#  define MPROTECT_VDB
```
Comment (ivmai) [inline include/private/gcconfig.h:1086]: If MPROTECT_VDB is no longer defined, then there is no need of Serenity-specific changes in code inside ifdef MPROTECT_VDB: i.e. please remove defined(SERENITY) in os_dep.c line 3263 and 3381.

**Comment by ivmai** [comment]
Okay, looks good, except for the comment regarding MPROTECT_VDB! I will process the PR in several day.
Meanwhile, I recommend you to go ahead and modify https://github.com/SerenityOS/serenity/blob/master/Ports/bdwgc/patches/0001-Teach-os_dep-and-gcconfig.h-about-serenity.patch

File: include/private/gcconfig.h (hunk 1 above)
Comment (linusg) [reply include/private/gcconfig.h:1086]: Thanks, updated!

**Comment by linusg** [comment]
> Meanwhile, I recommend you to go ahead and modify

I would prefer the next port update to be dropping serenity's patches altogether, happy to wait for this PR to be merged first :)

**Comment by ivmai** [comment]
Merged. Please test.

**Pull request closed without merging**